### PR TITLE
Disputes revised

### PR DESCRIPTION
**Pull request by adamaltman**

### CODE REVIEW

File: spec/definitions/Dispute.yaml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  evidence:
+    allOf:
+      - $ref: "#/definitions/DisputeEvidence"
+  evidenceSubmissionTime:
+    description: Dispute evidence submission time
+    type: string
+    format: date-time
+  resolvedTime:
+    description: Dispute resolved time
+    type: string
```
Comment (XaosSintez) [inline spec/definitions/Dispute.yaml]: We can use `ServerTimestamp` here

File: spec/definitions/Dispute.yaml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      - subscription-canceled
+      - uncategorized
+  type:
+    description: The dispute's type
+    type: string
+    enum:
+      - RET
+      - 1CB
+      - 2CB
+      - ARB
```
Comment (Zaphell) [inline spec/definitions/Dispute.yaml]: How about this to keep consistent with enum value formats:
```Values: {"information-request", "first-chargeback", "second-chargeback", "arbitration"}```

File: spec/definitions/Dispute.yaml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    type: string
+  transactionId:
+    description: The dispute's transaction ID
+    type: string
+  currency:
+    description: The dispute currency ISO Alpha code
+    type: string
+  amount:
+    description: The dispute amount
+    type: integer
```
Comment (slavcodev) [inline spec/definitions/Dispute.yaml]: According openapi spec, should be [`number`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types)

Comment (adamaltman) [inline spec/definitions/Dispute.yaml]: indeed... I copy/pasted this from another definition, so we probably have this mistake elsewhere too

File: spec/definitions/Dispute.yaml (hunk 1 above)
Comment (adamaltman) [inline spec/definitions/Dispute.yaml]: We can't because `ServerTimestamp` is `read-only`.

File: spec/paths/disputes@{id}.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          $ref: "#/headers/Rate-Limit-Limit"
+        Rate-Limit-Remaining:
+          $ref: "#/headers/Rate-Limit-Remaining"
+        Rate-Limit-Reset:
+          $ref: "#/headers/Rate-Limit-Reset"
+      schema:
+        $ref: "#/definitions/Dispute"
+    401:
+      $ref: "#/responses/AccessForbidden"
+    404:
```
Comment (slavcodev) [inline spec/paths/disputes@{id}.yaml]: **Not found** for `PUT`?

**Comment by slavcodev** [comment]
Looks fine

**Comment by adamaltman** [comment]
It didn't seem like this built to master...

**Comment by slavcodev** [comment]
@adamaltman what do you mean?

**Comment by RomanGotsiy** [comment]
@slavcodev for some reason the changes were not accounted in the master build (http://rebilly.github.io/RebillyAPI/swagger.json didn't contain them). i'm not sure why yet but suspect concurrent builds issue.

I've just rerun Travis job and now it is ok.

**Pull request merged**